projects/packages/search/src/class-module-control.php

```diff
@@ -61,94 +59,18 @@ public function is_instant_search_enabled() {
 	 * Activiate Search module
 	 */
 	public function activate() {
-		/**
-		 * Fires before a module is activated.
-		 *
-		 * @since 2.6.0
-		 *
-		 * @param string $module Module slug.
-		 * @param bool $exit Should we exit after the module has been activated. Default to true.
-		 * @param bool $redirect Should the user be redirected after module activation? Default to true.
-		 */
-		do_action( 'jetpack_pre_activate_module', self::JETPACK_SEARCH_MODULE_SLUG );
-
-		// If it's already active, then don't do it again.
-		if ( $this->is_active() ) {
-			return true;
-		}
-		// Not available for offline mode.
-		$is_offline_mode = ( new Status() )->is_offline_mode();
-		if ( $is_offline_mode ) {
```

Comment:
I'm wondering how Search plugin would claim it needs connection by using the `Modules` class?

Reply:
This doesn't seems to matter a lot for standalone plugins tho, since if site is not connected, the search dashboard wouldn't be shown anyway.